File: public/usage-examples/graphics/get_font_style_name_as_string-1-example.cpp
#include "splashkit.h"

int main()
{
    open_window("Interactive Font Style Changer", 800, 120);
    load_font("Arial", "Arial.TTF");

    // Initialise Default message
    string message = "Press N for Normal, B for Bold, I for Italics, or U for Underlined.";
    string message1 = "";
    font_style style = NORMAL_FONT;

    while (!quit_requested())
    {
        process_events();

        // Check key presses and update font style
        if (key_typed(N_KEY))
        {
            set_font_style("Arial", NORMAL_FONT);
        }
        else if (key_typed(B_KEY))
        {
            set_font_style("Arial", BOLD_FONT);
        }
        else if (key_typed(I_KEY))
        {
            set_font_style("Arial", ITALIC_FONT);
        }
        else if (key_typed(U_KEY))
        {
            set_font_style("Arial", UNDERLINE_FONT);
        }

        message1 = "Font style set to ";
        style = get_font_style("Arial");

        // A switch case is needed for c++ as get_font_style returns an enum not a string
        switch (style)
        {
        case NORMAL_FONT:
            message1 += "Normal";
            break;
        case BOLD_FONT:
            message1 += "Bold";
            break;
        case ITALIC_FONT:
            message1 += "Italic";
            break;
        case UNDERLINE_FONT:
            message1 += "Underlined";
            break;
        default:
            message1 += "Unknown";
            break;
        }

        // Clear screen and draw updated message
        clear_screen(COLOR_WHITE);
        draw_text(message, COLOR_BLACK, "Arial", 20, 50, 20);
        draw_text(message1, COLOR_BLACK, "Arial", 20, 50, 80);
        refresh_screen();
    }

    delay(5000);
    close_all_windows();
    return 0;
}

File: public/usage-examples/graphics/get_font_style_name_as_string-1-example.py
from splashkit import *

open_window("Font Style", 800, 120)

# Load font
# Unlike C++/C#, Python SplashKit requires using the loaded font object directly.
# Using the font name string like "Arial" will result in invalid font errors.
# To fix this, we load the font and keep a reference to the font object (e.g., `loaded_font`)
loaded_font = load_font("Arial", "Arial.TTF")

# Default message
InitialText = "Press N for Normal, B for Bold, I for Italics, or U for Underlined."
FontText = ""


while not quit_requested():
    process_events()

    # Check key presses and update font style
    if key_typed(KeyCode.n_key):
        set_font_style(loaded_font, FontStyle.normal_font)
    elif key_typed(KeyCode.b_key):
        set_font_style(loaded_font, FontStyle.bold_font)
    elif key_typed(KeyCode.i_key):
        set_font_style(loaded_font, FontStyle.italic_font)
    elif key_typed(KeyCode.u_key):
        set_font_style(loaded_font, FontStyle.underline_font)
    
    FontText = "Font style set to "
    style = get_font_style(loaded_font)
    if style == FontStyle.normal_font:
        FontText += "Normal"
    elif style == FontStyle.bold_font:
        FontText += "Bold"
    elif style == FontStyle.italic_font:
        FontText += "Italic"
    elif style == FontStyle.underline_font:
        FontText += "Underlined"
    else:
        FontText += "Unknown"
    
    # Clear screen and draw updated message
    clear_screen_to_white()
    draw_text(InitialText, color_black(), loaded_font, 20, 50, 20)
    draw_text(FontText, color_black(), loaded_font, 20, 50, 80)
    # Refresh the window with the updated text
    refresh_screen()  

delay(5000)
close_all_windows()